api/backups/restore_test.go

```diff
@@ -0,0 +1,48 @@
+// Copyright 2018 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package backups_test
+
+import (
+	//	jc "github.com/juju/testing/checkers"
+	gc "gopkg.in/check.v1"
+
+	"github.com/golang/mock/gomock"
+	"github.com/juju/juju/api/backups"
+
+	"github.com/juju/juju/api/base/mocks"
+	"github.com/juju/juju/apiserver/params"
+)
+
+type restoreSuite struct {
+}
+
+var _ = gc.Suite(&restoreSuite{})
+
+func (s *restoreSuite) TestRestoreFromFileBackupExistsOnController(c *gc.C) {
+	mockController := gomock.NewController(c)
+	mockBackupFacadeCaller := mocks.NewMockFacadeCaller(mockController)
+	mockBackupClientFacade := mocks.NewMockClientFacade(mockController)
+	mockBackupClientFacade.EXPECT().Close().AnyTimes()
+	testBackupResults := params.BackupsMetadataResult{
+		Checksum: "testCheckSum",
+	}
+	testBackupsListResults := params.BackupsListResult{
+		List: []params.BackupsMetadataResult{testBackupResults},
+	}
+	resultBackupList := &params.BackupsListResult{}
+	args := params.BackupsListArgs{}
+
+	gomock.InOrder(
+		mockBackupFacadeCaller.EXPECT().FacadeCall("PrepareRestore", nil, gomock.Any()),
+		mockBackupFacadeCaller.EXPECT().FacadeCall("List", args, resultBackupList).SetArg(2, testBackupsListResults),
+		mockBackupFacadeCaller.EXPECT().FacadeCall("Restore", gomock.Any(), gomock.Any()),
+		mockBackupFacadeCaller.EXPECT().FacadeCall("FinishRestore", gomock.Any(), gomock.Any()),
+	)
```

Comment:
Notice here how we do not make a call to Upload. Any call to upload would break this test.